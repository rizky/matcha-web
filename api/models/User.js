import db from '../db/database'

var table = 'user'

var User = {
	getAllUsers: (callback) => {
		return db.query(`Select * from ${table}`, callback)
	},

	getUserById: (id,callback) => {
		return db.query(`select * from ${table} where Id=?`, [id], callback)
	},

	addUser: (User,callback) => {
		console.log("inside service")
		console.log(User.Id)
		return db.query(`Insert into ${table} values(?,?,?)`, [User.Id, User.Username, User.Name],  callback)
	},

	deleteUser: (id,callback) => {
		return db.query(`delete from ${table} where Id=?`, [id], callback)
	},

	updateUser: (id,User,callback) => {
		return  db.query(`update ${table} set Title=?,Status=? where Id=?`, [User.Username, User.Name,id], callback)
	},

	deleteAll: (item,callback) => {
		var delete_array=[]
		for(i=0; i < item.length ; i++)
			delete_array[i] = item[i].Id
		return db.query(`delete from ${table} where Id in (?)`, [delete_array], callback)
	}
}

export default User