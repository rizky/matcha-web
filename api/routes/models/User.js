import ORM from './ORM'

export default class User extends ORM {
}