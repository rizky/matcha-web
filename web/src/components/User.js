import React, { Component } from 'react'
import './Photo.css'

export default class User extends Component {
	render () {
		if (!this.props.user)
			return null
		var lat = 46.529
		var long = 6.5
		let user = this.props.user
		let age = this.calculateAge(Date.parse(user.dob))
		let activeAt = this.timeSince(Date.parse(user.activeAt))
		let distance = this.getDistanceFromLatLonInKm(user.lat, user.long, lat, long)
		return (
			<div className='photo' id={`user_${user.id}`}>
				<img className='picture' src={user.picture} alt=''/>
				<div>
					<a>
						<span className='user'>{user.name}, </span>
						<span>{age}</span>
					</a>
					<i className='fa fa-ellipsis-v' style={{ float: 'right' }} onClick={this.handleShow}></i>
				</div>
				<div>
					<span className='time_text'>{distance} kilometer away, </span>
					<span className='time_text'> active {activeAt}</span>
				</div>
			</div>
		)
	}

	calculateAge(date) {
		var seconds = Math.floor((new Date() - date) / 1000)
		var interval = Math.floor(seconds / 31536000)
		return interval;
	}

	timeSince(date) {
		var seconds = Math.floor((new Date() - date) / 1000)
		var interval = Math.floor(seconds / 31536000)
		
		if (interval > 1) {
			return interval + " years ago"
		}
		interval = Math.floor(seconds / 2592000)
		if (interval > 1) {
			return interval + " months ago"
		}
		interval = Math.floor(seconds / 86400)
		if (interval > 1) {
			return interval + " days ago"
		}
		interval = Math.floor(seconds / 3600)
		if (interval > 1) {
			return interval + " hours ago"
		}
		interval = Math.floor(seconds / 60)
		if (interval > 1) {
			return interval + " minutes ago"
		}
		return Math.floor(seconds) + " seconds ago"
	}

	getDistanceFromLatLonInKm(lat1,lon1,lat2,lon2) {
		var R = 6371; // Radius of the earth in km
		var dLat = this.deg2rad(lat2-lat1);  // deg2rad below
		var dLon = this.deg2rad(lon2-lon1); 
		var a = 
			Math.sin(dLat/2) * Math.sin(dLat/2) +
			Math.cos(this.deg2rad(lat1)) * Math.cos(this.deg2rad(lat2)) * 
			Math.sin(dLon/2) * Math.sin(dLon/2)
			; 
		var c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1-a)); 
		var d = R * c; // Distance in km
		return Math.ceil(d)
	}
	
	deg2rad(deg) {
		return deg * (Math.PI/180)
	}
}