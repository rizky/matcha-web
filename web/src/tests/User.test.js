import React from 'react';
import User from '../components/User';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

test('User is NULL', () => {
	let user = null
	const component = shallow(<User user={user} />);
	expect(component.type()).toEqual(null);

	const display = renderer.create(
		<User user={user} />,
	  );
	let tree = display.toJSON();
 	expect(tree).toMatchSnapshot();
});

test('User is valid', () => {
	let user = {
		activeAt:"2018-05-30T16:39:15.000Z",
		createdAt:"2018-05-30T16:39:15.000Z",
		dob:"1988-05-30T16:39:15.000Z",
		email:"camagru.rizky@gmail.com",
		id:"0000000001",
		lat:46.528999,
		long:6.5626,
		name:"Admin",
		password:"3bb37061e887baa3b48ebe9f060f1a42baf995fb",
		picture:"/img/profiles/no-pic.jpg",
		subscribed:{type: "Buffer", data: Array(1)},
		tokenLost:null,
		tokenValidated:null,
		updatedAt:null,
		username:"admin",
	}
	const component = shallow(<User user={user} />);
	expect(component.length).toEqual(1);
	expect(component.find('.user').text()).toEqual(user.name+', ');

	const display = renderer.create(
		<User user={user} />,
	  );
	let tree = display.toJSON();
 	expect(tree).toMatchSnapshot();
});

test('Distance calculation', null);
test('Active time calculation', null);
test('Show more option', null);
test('Unmatch', null);
test('Like', null);
test('Pass', null);