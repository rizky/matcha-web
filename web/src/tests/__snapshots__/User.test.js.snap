// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User is NULL 1`] = `null`;

exports[`User is valid 1`] = `
<div
  className="photo"
  id="user_0000000001"
>
  <img
    alt=""
    className="picture"
    src="/img/profiles/no-pic.jpg"
  />
  <div>
    <a>
      <span
        className="user"
      >
        Admin
        , 
      </span>
      <span>
        30
      </span>
    </a>
    <i
      className="fa fa-ellipsis-v"
      onClick={undefined}
      style={
        Object {
          "float": "right",
        }
      }
    />
  </div>
  <div>
    <span
      className="time_text"
    >
      5
       kilometer away, 
    </span>
    <span
      className="time_text"
    >
       active 
      2 days ago
    </span>
  </div>
</div>
`;
